presto-main/src/main/java/com/facebook/presto/sql/rewrite/MaterializedViewOptimizationRewriteUtils.java

```diff
@@ -50,9 +55,18 @@ public static Query optimizeQueryUsingMaterializedView(
         for (QualifiedObjectName candidate : materializedViewCandidates) {
             Query optimizedQuery = getQueryWithMaterializedViewOptimization(metadata, session, sqlParser, accessControl, node, candidate);
             if (node != optimizedQuery) {
-                MaterializedViewStatus materializedViewStatus = metadata.getMaterializedViewStatus(session, candidate);
-                if (materializedViewStatus.isFullyMaterialized() || materializedViewStatus.isPartiallyMaterialized()) {
-                    session.getRuntimeStats().addMetricValue(OPTIMIZED_WITH_MATERIALIZED_VIEW, 1);
+                if (isMaterializedViewDataConsistencyEnabled(session)) {
+                    //TODO: We should be able to leverage this information in the StatementAnalyzer as well.
+                    MaterializedViewStatus materializedViewStatus = metadata.getMaterializedViewStatus(session, candidate);
+                    if (materializedViewStatus.isFullyMaterialized() || materializedViewStatus.isPartiallyMaterialized()) {
+                        session.getRuntimeStats().addMetricValue(OPTIMIZED_WITH_MATERIALIZED_VIEW, 1);
+                        return optimizedQuery;
+                    }
+                    else {
```

Comment:
`else` is redundant here

Reply:
Not sure if it is? We do not want to optimize the query using materialized view, if materialized view is missing too many partitions.